networkx/algorithms/isomorphism/tests/test_ismags.py

```diff
@@ -208,48 +215,134 @@ def test_labeled_edges(self):
         )
 
 
+def test_noncomparable_nodes():
+    node1 = object()
+    node2 = object()
+    node3 = object()
+
+    # Graph
+    G = nx.path_graph([node1, node2, node3])
+    gm = iso.ISMAGS(G, G)
+    assert gm.is_isomorphic()
+    # Just testing some cases
+    assert gm.subgraph_is_isomorphic()
+
+    # DiGraph
+    G = nx.path_graph([node1, node2, node3], create_using=nx.DiGraph)
+    H = nx.path_graph([node3, node2, node1], create_using=nx.DiGraph)
+    dgm = iso.ISMAGS(G, H)
+    assert dgm.is_isomorphic()
+    assert dgm.is_isomorphic(symmetry=True)
+    # Just testing some cases
+    assert dgm.subgraph_is_isomorphic()
+
+
+@pytest.mark.parametrize("graph_constructor", graph_classes)
+def test_selfloop(graph_constructor):
+    # Simple test for graphs with selfloops
+    g1 = graph_constructor([(0, 1), (0, 2), (1, 2), (1, 3), (2, 2), (2, 4)])
+    nodes = range(5)
+    rng = random.Random(42)
+
+    for _ in range(3):
+        new_nodes = list(nodes)
+        rng.shuffle(new_nodes)
+        d = dict(zip(nodes, new_nodes))
+        g2 = nx.relabel_nodes(g1, d)
+        assert iso.ISMAGS(g1, g2).is_isomorphic()
+
+
 class TestWikipediaExample:
+    # example in wikipedia is g1a and g2b
+    # 1 have letter nodes, 2 have number nodes
+    # b have some edges reversed vs a (undirected still isomorphic)
+    # reversed edges marked with blank comment `#`
+    # isomorphism = {'a': 1, 'g': 2, 'b': 3, 'c': 6, 'h': 4, 'i': 5, 'j': 7, 'd': 8}
+
     # Nodes 'a', 'b', 'c' and 'd' form a column.
     # Nodes 'g', 'h', 'i' and 'j' form a column.
-    g1edges = [
+    g1a_edges = [
+        ["a", "g"],
+        ["a", "h"],  #
+        ["a", "i"],
+        ["b", "g"],  #
+        ["b", "h"],
+        ["b", "j"],
+        ["c", "g"],  #
+        ["c", "i"],  #
+        ["c", "j"],
+        ["d", "h"],  #
+        ["d", "i"],
+        ["d", "j"],  #
+    ]
+    g1b_edges = [
         ["a", "g"],
-        ["a", "h"],
+        ["h", "a"],  #
```

Comment:
What's the point of these "comments"?

Reply:
A comment above states the # symbol indicates an edge with direction swapped between "a" and "b" graphs. I added words to the comments to indicate that.